zebra-chain/src/transparent/script.rs

```diff
@@ -12,7 +12,20 @@ use std::{fmt, io};
     any(test, feature = "proptest-impl"),
     derive(proptest_derive::Arbitrary)
 )]
-pub struct Script(pub Vec<u8>);
+pub struct Script(Vec<u8>);
+
+impl Script {
+    /// Create a new Bitcoin script from its raw bytes.
+    /// The raw bytes must not contain the compact size prefix.
+    pub fn new(raw_bytes: Vec<u8>) -> Self {
+        Script(raw_bytes)
+    }
+
+    /// Return the raw bytes of the script without the compact size prefix.
+    pub fn raw_bytes(&self) -> &Vec<u8> {
```

Comment:
We might want to call this `as_raw_bytes` to follow the Rust API Guidelines? https://rust-lang.github.io/api-guidelines/naming.html#c-conv

Reply:
Seems good to me, let's explicitly add a correctness note for serialization?

```rust
/// # Correctness
///
/// These raw bytes do not have a length prefix.
///
/// The Zcash serialization format requires a length prefix, use `zcash_serialize`
/// and `zcash_deserilize` to create byte data with a length prefix.
```

Reply:
Great suggestion! Changed the name and added the comment.